Source/Initialization/WarpXInitData.cpp

```diff
@@ -1200,3 +1216,205 @@ void WarpX::InitializeEBGridData (int lev)
+        auto extent = E_x.getExtent();
+        double unit = E_x.unitSI();
+        auto d = E.gridSpacing<double>();
+
+std::cout << "offset: " << offset[0] << " " << offset[1] << " " << offset[2] << "\n";
+std::cout << "extent: " << extent[0] << " " << extent[1] << " " << extent[2] << "\n";
+
+        unsigned long const ix_lo = floor( std::abs(x_lo-offset[0])/d[0] );
+        unsigned long const ix_hi = floor( std::abs(x_hi-offset[0])/d[0] );
+        unsigned long const iy_lo = floor( std::abs(y_lo-offset[1])/d[1] );
+        unsigned long const iy_hi = floor( std::abs(y_hi-offset[1])/d[1] );
+        unsigned long const iz_lo = floor( std::abs(z_lo-offset[2])/d[2] );
+        unsigned long const iz_hi = floor( std::abs(z_hi-offset[2])/d[2] );
+
+std::cout << "ix,y,z_lo: " << ix_lo << " " << iy_lo << " " << iz_lo << "\n";
+std::cout << "ix,y,z_hi: " << ix_hi << " " << iy_hi << " " << iz_hi << "\n";
+std::cout << "d: " << d[0] << " " << d[1] << " " << d[2] << "\n";
+std::cout << "ix_hi-ix_lo: " << ix_hi-ix_lo << " " << iy_hi-iy_lo << " " << iz_hi-iz_lo << "\n";
+
+        openPMD::Offset chunk_offset = {ix_lo, iy_lo, iz_lo};
+        openPMD::Extent chunk_extent = {ix_hi-ix_lo, iy_hi-iy_lo, ix_hi-iz_lo};
+
+        std::cout << "offset[]+extent[]*d[]: "
+                  << offset[0]+extent[0]*d[0] << " "
+                  << offset[1]+extent[1]*d[1] << " "
+                  << offset[2]+extent[2]*d[2] << "\n";
+        std::cout << "d: " << d[0] << " " << d[1] << " " << d[2] << "\n";
+        auto Ex_chunk_data = E_x.loadChunk<double>(chunk_offset, chunk_extent);
+        series.flush();
+
+        //// Obtain Array4 from FArrayBox.
+        //auto arr = F1.array(mfi);
+
+        amrex::ParallelFor (tbx, tby, tbz,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+
+                amrex::Real x, y, z;
+
+                if ( box.type(0)==1 ) { x = real_box.lo(0) + i*dx[0]; }
+                else { x = real_box.lo(0) + i*dx[0] + 0.5*dx[0]; }
+                if ( box.type(1)==1 ) { y = real_box.lo(1) + j*dx[1]; }
+                else { y = real_box.lo(1) + j*dx[1] + 0.5*dx[1]; }
+                if ( box.type(2)==1 ) { z = real_box.lo(2) + k*dx[2]; }
+                else { z = real_box.lo(2) + k*dx[2] + 0.5*dx[2]; }
+
+                int ix = floor( std::abs(x-offset[0])/d[0] );
+                int iy = floor( std::abs(y-offset[1])/d[1] );
+                int iz = floor( std::abs(z-offset[2])/d[2] );
+                amrex::Real xx, yy, zz, ddx, ddy, ddz;
+                xx = offset[0] + ix*d[0];
+                yy = offset[1] + iy*d[1];
+                zz = offset[2] + iz*d[2];
+                ddx = (xx-x)/d[0];
+                ddy = (yy-y)/d[1];
+                ddz = (zz-z)/d[2];
+                int ext_1  = chunk_extent[1];
+                int ext_12 = chunk_extent[1]*chunk_extent[2];
+
+                //mfxfab(i,j,k) = Ex_chunk_data[(iz  )+(iy  )*ext_1+(ix  )*ext_12]*(1.0-ddx)*(1.0-ddy)*(1.0-ddz) +
```

Comment:
`Ex_chunk_data` is a `std::shared_ptr<T>`, you need to call `.get()` on it to get the data pointer:
https://en.cppreference.com/w/cpp/memory/shared_ptr/get

Store this pointer in a raw pointer before the parallel loop, similar to as we do the `Array4` array access before `ParallelFor`.